modules/azure-firewall/inbound.tf

```diff
@@ -0,0 +1,43 @@
+data "azurerm_public_ip" "pip" {
+  count               = length(var.firewall)
+  name                = azurerm_public_ip.main[count.index].name
+  resource_group_name = azurerm_public_ip.main[count.index].resource_group_name
+}
+
+resource "azurerm_firewall_nat_rule_collection" "main" {
+  count               = length(var.firewall)
+  name                = "dnatRule-${lookup(element(var.firewall, count.index), "location")}"
```

Comment:
shouldn't this come from configuration too? (the name)

Reply:
it totally depends on what name we would like to see here. For the current implementation, the name will be formed taking the values from variables file and is there depending on the region the firewall is implemented.

Reply:
Hence if the firewall is for uksouth then the rules are created with the name dnaRule-uksouth